routers/web/web.go

```diff
@@ -757,6 +757,7 @@ func RegisterRoutes(m *web.Route) {
 				m.Post("/reactions/{action}", bindIgnErr(forms.ReactionForm{}), repo.ChangeIssueReaction)
 				m.Post("/lock", reqRepoIssueWriter, bindIgnErr(forms.IssueLockForm{}), repo.LockIssue)
 				m.Post("/unlock", reqRepoIssueWriter, repo.UnlockIssue)
+				m.Post("/delete", reqRepoIssueWriter, repo.DeleteIssue)
```

Comment:
I suppose the reqRepoIssueWriter sets the correct permissions, as it's also used for the Locking route. But the name sounds weird.

Also in terms of allowing users to do the same for PRs which is an explicit or requirement for certain routes.